tests/versioned/apollo-server/segments.test.js

```diff
@@ -5,9 +5,33 @@
 
 'use strict'
 
-const { setupApolloServerTests } = require('./apollo-server-setup')
-const expressSegmentsDefaultTests = require('../express-segments-default-tests')
-const expressSegmentsScalarTests = require('../express-segments-scalar-tests')
+const test = require('node:test')
 
-setupApolloServerTests(expressSegmentsDefaultTests)
-setupApolloServerTests(expressSegmentsScalarTests)
+const { afterEach, setupCoreTest } = require('../../test-tools')
+
+const expressSegmentsTests = require('../express-segments-tests')
+
+testDefaultTests(expressSegmentsTests.tests).then(() => testScalarTests(expressSegmentsTests.tests))
```

Comment:
can't you just remove the async functions and run all of the test files like this:

```js
diff --git a/tests/versioned/apollo-server/segments.test.js b/tests/versioned/apollo-server/segments.test.js
index d6e0d50..2ec915c 100644
--- a/tests/versioned/apollo-server/segments.test.js
+++ b/tests/versioned/apollo-server/segments.test.js
@@ -11,27 +11,21 @@ const { afterEach, setupCoreTest } = require('../../test-tools')
 
 const expressSegmentsTests = require('../express-segments-tests')
 
-testDefaultTests(expressSegmentsTests.tests).then(() => testScalarTests(expressSegmentsTests.tests))
-
-async function testDefaultTests(tests) {
-  for (const defTest of tests) {
-    test(`non-scalar: ${defTest.name}`, async (t) => {
-      await setupCoreTest({ t, testDir: __dirname })
-      t.nr.TRANSACTION_PREFIX = 'WebTransaction/Expressjs/POST'
-      await defTest.fn(t)
-      await afterEach({ t, testDir: __dirname })
-    })
-  }
+for (const defTest of expressSegmentsTests.tests) {
+  test(`non-scalar: ${defTest.name}`, async (t) => {
+    await setupCoreTest({ t, testDir: __dirname })
+    t.nr.TRANSACTION_PREFIX = 'WebTransaction/Expressjs/POST'
+    await defTest.fn(t)
+    await afterEach({ t, testDir: __dirname })
+  })
 }
 
-async function testScalarTests(tests) {
-  const { pluginConfig } = { captureScalars: true }
-  for (const scalarTest of tests) {
-    test(`scalar: ${scalarTest.name}`, async (t) => {
-      await setupCoreTest({ t, testDir: __dirname })
-      t.nr.TRANSACTION_PREFIX = 'WebTransaction/Expressjs/POST'
-      await scalarTest.fn(t)
-      await afterEach({ t, testDir: __dirname, pluginConfig })
-    })
-  }
+const { pluginConfig } = { captureScalars: true }
+for (const scalarTest of expressSegmentsTests.tests) {
+  test(`scalar: ${scalarTest.name}`, async (t) => {
+    await setupCoreTest({ t, testDir: __dirname })
+    t.nr.TRANSACTION_PREFIX = 'WebTransaction/Expressjs/POST'
+    await scalarTest.fn(t)
+    await afterEach({ t, testDir: __dirname, pluginConfig })
+  })
 }
```

I modified this one locally and runs as expected. it's very confusing to read this unnecessary promise chain in all test files

Reply:
For these suites of multiple versions (e.g. both `true` and `false` cases), I was seeing errors about the agent being created multiple times. Wrapping them up as I have done resolved this issue.

Reply:
I think that's because of the other issue I'm suggesting addressing though